examples/mnist_advanced_gans/callbacks.py

```diff
@@ -0,0 +1,492 @@
+"""
+All custom callbacks
+"""
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+import torchvision.utils
+
+from catalyst.dl import RunnerState
+from catalyst.dl.callbacks import CriterionCallback, OptimizerCallback
+from catalyst.dl.core import Callback, CallbackOrder
+from catalyst.utils.tensorboard import SummaryWriter
+"""
+InputBatchTransform functions,
+e.g.:
+- add noise to batch: state.input["noise"] = z
+- add real/fake targets (zeros/ones)
+- transform targets to/from one-hot encoding
+Note: these transforms are different from dataset transforms as they transform
+the entire batch which may be more efficient/simple than custom batch_collate
+"""
+
+
+class InputBatchTransformCallback(Callback):
```

Comment:
is it possible to make all this logic with PyTorch Transforms, Samplers and Datasets?

Reply:
or with the usage of [mixins](https://github.com/catalyst-team/catalyst/tree/master/catalyst/contrib/mixin)?